### PR TITLE
Adding Biased Upwind Operator

### PR DESCRIPTION
**Pull request by mjsheikh**
Implementation of a Biased Upwind Operator ( #97 )
TODO : Doc changes once the code gets approved.

### CODE REVIEW

File: src/derivative_operators/derivative_operator.jl [hunk 1]
```diff
@@ -182,69 +187,85 @@ julia> Array(L2 * Q)[1]
...
-        high_boundary_coefs,coefficients,coeff_func
+        high_boundary_coefs,offside,coefficients,coeff_func
         )
 end
 
 # TODO implement the non-uniform grid
 function UpwindDifference{N}(derivative_order::Int,
                           approximation_order::Int, dx::AbstractVector{T},
-                          len::Int, coeff_func=nothing) where {T<:Real,N}
+                          len::Int, offside::Int, coeff_func=nothing) where {T<:Real,N}
```
Comment (ChrisRackauckas) [inline src/derivative_operators/derivative_operator.jl:233]: should be a keyword argument

**Comment by mjsheikh** [comment]
Will add tests shortly

File: test/heat_eqn.jl [hunk 2]
```diff
@@ -33,7 +33,7 @@ end
     dx = 2π/(N-1)
     x = collect(-pi : dx : pi)
     u0 = @. -(x - 0.5)^2 + 1/12
-    B = CenteredDifference(1,2,dx,N)
+    B = CenteredDifference(1,2,dx,N-2)
```
Comment (ChrisRackauckas) [inline test/heat_eqn.jl:36]: why the size change?

Comment (mjsheikh) [reply test/heat_eqn.jl:36]: > why the size change?

Its operating on a Vector with 512 grid points(including boundaries).
That was working somehow on Centered difference but Upwind was returning dimension mismatch error.

**Pull request merged**